### ISSUE TITLE
Feature request for only converting only certain structs

### ISSUE THREAD

**Issue by JamesTiberiusKirk** [gzuidhof/tygo#11]

**Comment by JamesTiberiusKirk** [comment]
Sorry, tapped endter before I finished and it published it straight away.
First of all I wanna say that I love this package and its almost what im looking for!
But it might be a good idea to see if we can say only generate ts types for only certain structs.
Example: only for structs which match a name ```*DTO``` or ```*Data```.
or another Example: only for structs which have the json tags

**Comment by JamesTiberiusKirk** [comment]
If we agree I could even help out and open a pr for this.

**Comment by JamesTiberiusKirk** [comment]
Im playing around with the codebase at the moment.
Maying im being a bit daft but based on name it seems a bit hard todo.
Maybe another idea is to leave a comment on each struct that we want to include
something like ```TsType``` anywhere in the comment for the ast.Node

**Comment by gzuidhof** [comment]
I don't know what your current draft implementation looks like of course, but perhaps we can allow users to specify it in their config file instead of putting tags into the Go code?

Of course either can work

**Comment by JamesTiberiusKirk** [comment]
yes, so what I have done is added another field in the config yaml (```include_struct_comment```) where the user can specify a specific string to use.
if the string has been specified, then the user would need to add a new comment line to each and every struct they would want to be transpiled to ts.
I have modified one of the examples in the repo to shocase the feature

**Comment by gzuidhof** [comment]
I meant in the direction of:

```yaml
include_symbols:
  - "*Suffix" # a glob with wildcard
  - "SomeConstant"
```

**Comment by JamesTiberiusKirk** [comment]
Just to make sure im not missunderstanding...
And the symbols will be used as "tags" left by the user in the comments of the struct?

**Comment by gzuidhof** [comment]
I'm proposing that the user, in the config, specifies the structs / consts / types that they want present. Either with their exact names, or matching some regex. So there would be no special comments at all. What are your thoughts on that approach?

**Comment by JamesTiberiusKirk** [comment]
yeah i like that approach, but not gunna like im not sure how i can get access to the node name from *ast.GenDecl in package_generator.go

**Comment by JamesTiberiusKirk** [comment]
unless you can think of a better place of implementing this?

**Comment by gzuidhof** [comment]
I'm happy to have a look at it, but it will probably only be in the weekend :/. I imagine we can traverse the tree down a bit to find the name if we have to (if generation takes 1ms longer it's no problem).

**Comment by JamesTiberiusKirk** [comment]
ok great. let me know if theres anything else I can do

**Comment by JamesTiberiusKirk** [comment]
did you manage to have a look at all @gzuidhof ?

**Comment by gzuidhof** [comment]
I'm sorry, I haven't gotten around to it yet :(
